node/libs/crypto/src/bn254/mod.rs

```diff
@@ -25,52 +27,66 @@ pub mod hash;
 mod testonly;
 
 /// Type safety wrapper around a scalar value.
+#[derive(Debug, PartialEq)]
```

Comment:
Don't implement PartialEq for private key. Always compare corresponding public keys instead.

Reply:
What about this: https://github.com/matter-labs/era-consensus/pull/28/commits/e3074cbee597d0c722dc005d0f4370d22003a14e?